libraries/launcher/org/prismlauncher/launcher/impl/StandardLauncher.java

```diff
@@ -97,6 +97,18 @@ public void launch() throws Throwable {
             gameArgs.add(worldName);
         }
 
+        StringBuilder joinedGameArgs = new StringBuilder();
+        for (String gameArg : gameArgs) {
+            if (joinedGameArgs.length() > 0) {
+                joinedGameArgs.append('\u001F'); // unit separator, designed for this purpose
+            }
+            joinedGameArgs.append(gameArg);
+        }
+
+        // pass the real main class and game arguments in so mods can access them
+        System.setProperty("org.prismlauncher.launch.mainclass", mainClassName);
+        System.setProperty("org.prismlauncher.launch.gameargs", joinedGameArgs.toString());
```

Comment:
```suggestion
        // unit separator ('\u001F') delimited list of game args
        System.setProperty("org.prismlauncher.launch.gameargs", joinedGameArgs.toString());
```

Reply:
This is already documented a few lines above when joining into the stringbuilder.